lib/spack/spack/directives.py

```diff
@@ -182,10 +192,28 @@ def _depends_on(pkg, spec, when=None):
         when = pkg.name
     when_spec = parse_anonymous_spec(when, pkg.name)
 
+    if deptypes is None:
+        # The default deptype is build and install because the common case is
+        # to build against a library which then turns into a runtime dependency
+        # due to the linker.
+        # XXX(deptype): Add DEP_RUN to this? It's an uncommon dependency type,
```

Comment:
The reason I kept "link" out of it is because of things like Java which don't *link*, but do need the jarpath set up similarly to find dependent `.jar` files (I think, not *that* familiar with Java). But I guess such things are similar enough to "link" that Java (packaging) folks would understand it.

As for `DEP_RUN`, the backwards compat is nice, but is /so/ rare that having `depends_on` mention it to *not* add a run dependency is noisy. Finding things which actually care about things like `dwarfdump` being in the `PATH` is easy and makes the `depends_on` lines a lot more informative.